src/pymor/reductors/basic.py

```diff
@@ -177,9 +180,16 @@ def project_operators(self):
         projected_operators = {
             'operator':          project(fom.operator, RB, RB),
             'rhs':               project(fom.rhs, RB, None),
-            'products':          {k: project(v, RB, RB) for k, v in fom.products.items()},
             'output_functional': project(fom.output_functional, None, RB)
         }
+
+        products = {k: project(v, RB, RB) for k, v in fom.products.items() if RB in v.source and RB in v.range}
+        if products:
+            projected_operators['products'] = products
+        else:
+            self.logger.warning('FOM has no products that are compatible with RB. '
+                                'ROM will use Euclidean inner product.')
```

Comment:
Sorry, but looking at the code, I think I've changed my mind. I think we should rather enforce that `products` of `StationaryModel` and `InstationaryModel` all act on `solution_space`.

What are the benefits of having a product in `products`?
- Products are automatically projected, but only when they match the RB.
- `Model` automatically adds `key_product` and `key_norm` attributes. The `key_norm` attributes are of limited use, I'd say. Most people probably just write `U.norm(product)`.
- You can write loops over all products. But if the act on different spaces, this is not really useful.

So overall, I think it is no practical limitation to restrict `products` to be products on `solution_space`, and it makes it easer to reason about them. Would you agree, @ullmannsven.